juriscraper/opinions/united_states/state/colo.py

```diff
@@ -23,25 +26,62 @@ def __init__(self, *args, **kwargs):
         self.url = "https://www.courts.state.co.us/Courts/Supreme_Court/Case_Announcements/"
 
     def _process_html(self) -> None:
-
         date = self.html.xpath(
             ".//p[contains(text(), 'case announcements for')]/text()"
-        )[0]
-        date_str = (
-            re.findall(r"case announcements for (.*) are now", date)[0]
-            .split(",", 1)[1]
-            .strip()
         )
-        for item in self.html.xpath(
-            ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
-        ):
-            docket, name = item.text_content().split(" ", 1)
-            self.cases.append(
-                {
-                    "date": date_str,
-                    "docket": docket,
-                    "name": name,
-                    "url": item.attrib["href"],
-                    "status": "Published",
-                }
+        if date:
+            date_str = (
+                re.findall(r"case announcements for (.*) are now", date[0])[0]
+                .split(",", 1)[1]
+                .strip()
             )
+            for item in self.html.xpath(
+                ".//a[contains(@href, 'Supreme_Court/Opinions/')]"
+            ):
+                link_text = normalize_dashes(item.text_content().strip())
+                cite_match = re.findall(r"(\d{2,4}\s*CO\s*\d+\w?)", link_text)
+                docket_match = re.findall(r"\d+[A-Z]+\d{2,}", link_text)
+                docket = docket_match[0] if docket_match else ""
+                citation = cite_match[0] if cite_match else ""
+                name = (
+                    link_text.replace(docket, "")
+                    .replace(citation, "")
+                    .lstrip(",- ")
+                )
+                self.cases.append(
+                    {
+                        "date": date_str,
+                        "docket": docket,
+                        "name": name,
+                        "url": item.attrib["href"],
+                        "status": "Published",
+                        "citation": citation,
+                    }
+                )
+
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Pass scraped text into function and return data as a dictionary
+        Notes for 'Citation':
+            - Reporter key for this court: 'CO'
+            - Type for a state: 2
+        :param scraped_text: Text of scraped content
+        :return: metadata
+        """
+        match_re = r"The Supreme Court of the State of Colorado.*?(?P<citation>\d{4}\s*CO\s*\d+).*?Supreme Court Case No\. (?P<docket>\d+S[A-Z]\d+)"
+        match = re.findall(match_re, scraped_text, re.M | re.S)
+        if not match:
+            metadata = {}
+        else:
+            volume, page = match[0][0].split("CO")
+            metadata = {
+                "OpinionCluster": {
+                    "docket_number": match[0][1],
+                },
```

Comment:
Not sure we need to grab the docket number if we've already grabbed the docket number.

Reply:
Because of cases like this week's opinion:
> [22CO7, 19SC763- People v. Ray Ojeda](https://www.courts.state.co.us/userfiles/file/Court_Probation/Supreme_Court/Opinions/2019/19SC763.pdf)

In this case, the citation is missing the year. It's just a precaution in case the docket might also be incomplete.

![Screenshot from 2022-02-16 09-21-50](https://user-images.githubusercontent.com/8507838/154296820-590be4fe-a871-4eb7-8379-c65eb52a2c1a.png)

![Screenshot from 2022-02-16 09-22-05](https://user-images.githubusercontent.com/8507838/154296855-98ec8c08-c971-43c0-b1b7-7c2b243950b8.png)